modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs

```diff
@@ -226,8 +250,16 @@ select ouRole.RoleId
             .WhereIf(!string.IsNullOrWhiteSpace(userName), x => x.UserName == userName)
             .WhereIf(!string.IsNullOrWhiteSpace(phoneNumber), x => x.PhoneNumber == phoneNumber)
             .WhereIf(!string.IsNullOrWhiteSpace(emailAddress), x => x.Email == emailAddress)
+            .WhereIf(!string.IsNullOrWhiteSpace(name), x => x.Name == name)
+            .WhereIf(!string.IsNullOrWhiteSpace(surname), x => x.Surname == surname)
             .WhereIf(isLockedOut == true, x => x.LockoutEnabled && x.LockoutEnd.Value.CompareTo(DateTime.UtcNow) > 0)
             .WhereIf(notActive == true, x => !x.IsActive)
+            .WhereIf(emailConfirmed == true, x => x.EmailConfirmed)
```

Comment:
this can be wrong.
Assume we want to filte user with email not confirmed. it will not work!
```suggestion
            .WhereIf(emailConfirmed.HasValue, x => x.EmailConfirmed)
```
for `isExternal` it has the same problem